htp/Makefile.am

```diff
@@ -27,5 +27,5 @@ libhtp_c_la_SOURCES = $(h_sources) $(h_sources_private) $(c_sources)
 
 lib_LTLIBRARIES = libhtp.la
 libhtp_la_SOURCES =
-libhtp_la_LIBADD = libhtp-c.la lzma/liblzma-c.la
-libhtp_la_LDFLAGS = -version-info $(GENERIC_LIBRARY_VERSION)
+libhtp_la_LIBADD = libhtp-c.la lzma/liblzma-c.la $(LTLIBICONV)
```

Comment:
Is this not supposed to be made somehow by configure.ac with `:LIBS="${LIBS} ${LIBICONV}"` ?

Reply:
I confirmed that it can be built by doing as noted above.

```diff
diff --git a/configure.ac b/configure.ac
index 1ed5e8d..3e0adfb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -171,7 +171,7 @@ AC_MSG_RESULT($ac_cv_func_iconvctl)
 if test "$ac_cv_func_iconvctl" == yes; then
     AC_DEFINE(HAVE_ICONVCTL,1,"Define to 1 if you have the `iconvctl' function.")
 fi
-LIBS="${TMPLIBS}"
+# LIBS="${TMPLIBS}"
 
 dnl -----------------------------------------------
 dnl Check and enable the GCC opts we want to use.
diff --git a/htp/Makefile.am b/htp/Makefile.am
index e467b4b..9ab2311 100644
--- a/htp/Makefile.am
+++ b/htp/Makefile.am
@@ -27,5 +27,5 @@ libhtp_c_la_SOURCES = $(h_sources) $(h_sources_private) $(c_sources)
 
 lib_LTLIBRARIES = libhtp.la
 libhtp_la_SOURCES =
-libhtp_la_LIBADD = libhtp-c.la lzma/liblzma-c.la $(LTLIBICONV)
+libhtp_la_LIBADD = libhtp-c.la lzma/liblzma-c.la
 libhtp_la_LDFLAGS = -version-info $(GENERIC_LIBRARY_VERSION) -no-undefined
```

Reply:
Why would we need to comment the line `LIBS="${TMPLIBS}"` ?

Reply:
Writing back `TMPLIBS` will cause `LTLIBICONV` to disappear from `LIBS`.

Reply:
This problem was solved with `LIBS=-liconv ./configure && make` (no need to patch)